### PR TITLE
Extract common APIs into separate modules

### PR DESCRIPTION
**Pull request by li-wjohnson**
Added two new modules:

- `test-butler-api`: contains the ButlerApi AIDL

- `test-butler-app-core`: (or should it be test-butler-app-common?) contains reusable ButlerService APIs. Most important additions are:
  - `SettingsAccessor`, which is an abstraction for modifying Settings without an explicit application context
  - `CommonDeviceLocks`, which encapsulates all the device locks needed by ButlerService. 
  - `ButlerApiStubBase`, which uses SettingsAccessor to implement most of the ButlerApi. 

  Most other classes are copied from `test-butler-app`, with slight modifications to use the `SettingsAccessor` abstraction.

ButlerService then becomes very simple -- create an Application-based SettingsAccessor and call the common APIs. The only methods not handled in the common code are WifiState, GsmState, and Permissions, which require an application context.

Also, notice that I removed the `startService` and `stopService` calls from the Test Butler library. AFAIK, binding is enough, and in more recent versions of Android, starting a background service like this isn't normally permitted.

This change is a precursor to adding a second TestButlerApp implementation.

### CODE REVIEW

File: test-butler-app-core/src/main/java/com/linkedin/android/testbutler/AlwaysFinishActivitiesChanger.java [hunk 1]
```diff
@@ -30,43 +27,48 @@
 
     private static final String TAG = AlwaysFinishActivitiesChanger.class.getSimpleName();
 
+    private final SettingsAccessor settings;
+
     private boolean originalAlwaysFinishActivitiesMode;
 
+    AlwaysFinishActivitiesChanger(SettingsAccessor settings) {
```
Comment (drewhannay) [inline test-butler-app-core/src/main/java/com/linkedin/android/testbutler/AlwaysFinishActivitiesChanger.java]: Can you make sure we have either `@NonNull` or `@Nullable` on any method parameters of method return types?

File: test-butler-library/src/main/java/com/linkedin/android/testbutler/TestButler.java [hunk 2]
```diff
@@ -108,7 +108,6 @@ public static void setup(@NonNull Context context) {
         intent.setComponent(new ComponentName("com.linkedin.android.testbutler",
                 "com.linkedin.android.testbutler.ButlerService"));
 
-        context.startService(intent);
```
Comment (drewhannay) [inline test-butler-library/src/main/java/com/linkedin/android/testbutler/TestButler.java]: @kkoser can you test this and make sure it still works for emulators with older API levels?

Comment (li-wjohnson) [reply test-butler-library/src/main/java/com/linkedin/android/testbutler/TestButler.java]: I tried running the demo on an API 16 emulator. Service binding worked fine -- 4 of 8 passed, and the 4 that failed were due to features that don't exist / aren't supported in API 16.

Comment (kkoser) [reply test-butler-library/src/main/java/com/linkedin/android/testbutler/TestButler.java]: I  was also able to verify this on api22 that the service was still started

**Pull request merged**